The self-service uses a hardcoded database host and credentials,
and a hardcoded memcached socket.

This makes the Postgres and Memcached connection settings configurable.

diff --git /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/__init__.py /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/__init__.py
index f7ce009d2c..d50f4b8f3b 100644
--- /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/__init__.py
+++ /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/__init__.py
@@ -36,6 +36,7 @@

 import datetime
 import email.charset
+import os
 import os.path
 import random
 import smtplib
@@ -72,7 +73,9 @@ from .tokendb import MultipleTokensInDB, TokenDB

 _ = Translation('univention-self-service-passwordreset-umc').translate

-MEMCACHED_SOCKET = "/var/lib/univention-self-service-passwordreset-umc/memcached.socket"
+MEMCACHED_SOCKET = os.getenv("SELF_SERVICE_MEMCACHED_SERVER", "/var/lib/univention-self-service-passwordreset-umc/memcached.socket")
+MEMCACHED_USERNAME = os.getenv("SELF_SERVICE_MEMCACHED_USER")
+MEMCACHED_SECRET_FILE = os.getenv("SELF_SERVICE_MEMCACHED_SECRET_FILE")
 MEMCACHED_MAX_KEY = 250

 SELFSERVICE_MASTER = ucr.get("self-service/backend-server", ucr.get("ldap/master"))
@@ -258,7 +261,16 @@ class Instance(Base):
             atexit.register(self.db.close_db)
             if not self.db.table_exists():
                 self.db.create_table()
-            self.memcache = pylibmc.Client([MEMCACHED_SOCKET], binary=True)
+            if MEMCACHED_SECRET_FILE:
+                try:
+                    with open(MEMCACHED_SECRET_FILE) as pw_file:
+                        password = pw_file.readline().strip()
+                except (OSError, IOError) as e:
+                    self.logger.error(f"Instance.init(): Could not read {MEMCACHED_SECRET_FILE}: {e}")
+                    raise
+            else:
+                password = None
+            self.memcache = pylibmc.Client([MEMCACHED_SOCKET], binary=True, username=MEMCACHED_USERNAME, password=password)

         self.send_plugins = get_sending_plugins(MODULE.process)
         self.password_reset_plugins = {k: v for k, v in self.send_plugins.items() if v.message_application() == 'password_reset'}
diff --git /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/sending/send_email.py /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/sending/send_email.py
index f8001c8d93..e52fc3140c 100644
--- /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/sending/send_email.py
+++ /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/sending/send_email.py
@@ -49,6 +49,7 @@
 #

 import email.charset
+import os
 import os.path
 import smtplib
 from email.mime.nonmultipart import MIMENonMultipart
@@ -69,6 +70,24 @@ class SendEmail(UniventionSelfServiceTokenEmitter):
     def __init__(self, *args, **kwargs):
         super(SendEmail, self).__init__(*args, **kwargs)
         self.server = self.ucr.get("umc/self-service/passwordreset/email/server", "localhost")
+        self.port = self.ucr.get_int("umc/self-service/passwordreset/email/server/port", 0)
+        self.user = self.ucr.get("umc/self-service/passwordreset/email/server/user")
+        self.ehlo = self.ucr.get("umc/self-service/passwordreset/email/server/ehlo")
+        self.starttls = self.ucr.is_true("umc/self-service/passwordreset/email/server/starttls")
+
+        if self.user:
+            secret_file = os.getenv("SMTP_SECRET_FILE")
+            try:
+                with open(secret_file, 'r') as fd:
+                    self.password = fd.readline().strip()
+            except IOError:
+                self.log("SMTP_SECRET_FILE (%s) could not be read." % secret_file)
+                raise
+
+        if (self.user or self.starttls) and not self.ehlo:
+            hostname = self.ucr.get('hostname')
+            domainname = self.ucr.get('domainname')
+            self.ehlo = f"{hostname}.{domainname}"

     @staticmethod
     def send_method():
@@ -109,6 +128,8 @@ class SendEmail(UniventionSelfServiceTokenEmitter):
         fqdn = ".".join([self.ucr["hostname"], self.ucr["domainname"]])
         frontend_server = self.ucr.get("umc/self-service/passwordreset/email/webserver_address", fqdn)
         links = {
+            'fqdn': fqdn,
+            'domainname': self.ucr["domainname"],
             'link': f"https://{frontend_server}/univention/selfservice/#/selfservice/newpassword/",
             'tokenlink': "https://{fqdn}/univention/selfservice/#/selfservice/newpassword/?token={token}&username={username}".format(fqdn=frontend_server, username=quote(self.data["username"]), token=quote(self.data["token"])),
         }
@@ -128,7 +147,13 @@ class SendEmail(UniventionSelfServiceTokenEmitter):
         msg["To"] = self.data["address"]
         msg.set_payload(txt, charset=cs)

-        smtp = smtplib.SMTP(self.server)
+        smtp = smtplib.SMTP(self.server, self.port)
+        if self.starttls:
+            smtp.ehlo(self.ehlo)
+            smtp.starttls()
+        if self.user:
+            smtp.ehlo(self.ehlo)
+            smtp.login(self.user, self.password)
         smtp.sendmail(msg["From"], self.data["address"], msg.as_string())
         smtp.quit()
         self.log("Sent mail with token to address {}.".format(self.data["address"]))

diff --git /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/tokendb.py /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/tokendb.py
index e4266b4ae8..842457fd8e 100644
--- /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/tokendb.py
+++ /usr/lib/python3/dist-packages/univention/management/console/modules/passwordreset/tokendb.py
@@ -35,15 +35,18 @@
 # <https://www.gnu.org/licenses/>.

 import datetime
+import os
 import traceback

 import psycopg2
 import psycopg2.extras


-DB_USER = "selfservice"
-DB_NAME = "selfservice"
-DB_SECRETS_FILE = "/etc/self-service-db.secret"
+DB_HOST = os.getenv("SELF_SERVICE_DB_HOST", "localhost")
+DB_PORT = os.getenv("SELF_SERVICE_DB_PORT", "5432")
+DB_USER = os.getenv("SELF_SERVICE_DB_USER", "selfservice")
+DB_NAME = os.getenv("SELF_SERVICE_DB_NAME", "selfservice")
+DB_SECRETS_FILE = os.getenv("SELF_SERVICE_DB_SECRET_FILE", "/etc/self-service-db.secret")


 class MultipleTokensInDB(Exception):
@@ -120,8 +123,8 @@ token VARCHAR(255) NOT NULL);""")
             self.logger.error(f"db_open(): Could not read {DB_SECRETS_FILE}: {e}")
             raise
         try:
-            conn = psycopg2.connect("dbname={db_name} user={db_user} host='localhost' password='{db_pw}'".format(
-                db_name=DB_NAME, db_user=DB_USER, db_pw=password))
+            conn = psycopg2.connect("dbname={db_name} user={db_user} host='{db_host}' port={db_port} password='{db_pw}'".format(
+                db_name=DB_NAME, db_user=DB_USER, db_pw=password, db_host=DB_HOST, db_port=DB_PORT))
             self.logger.info("db_open(): Connected to database '{}' on server with version {} using protocol version {}.".format(
                 DB_NAME, conn.server_version, conn.protocol_version))
             return conn
